unitary/examples/tictactoe/tic_tac_toe_test.py

```diff
@@ -142,16 +144,107 @@ def test_rulesets(run_on_hardware):
         board.move("ab", tictactoe.TicTacSquare.O)
 
 
-@pytest.mark.parametrize("run_on_hardware", [False, True])
-def test_print(run_on_hardware):
-    board = tictactoe.TicTacToe(run_on_hardware=run_on_hardware)
+def test_welcome():
+    board = tictactoe.TicTacToe()
+    game = tictactoe.GameInterface(board)
+    output = game.print_welcome()
+
+    assert (output == """
+        Welcome to quantum tic tac toe!
+        Here is the board:
+        
+        a | b | c
+        -----------
+        d | e | f
+        -----------
+        g | h | i
+        """
+    )
+
+def test_help():
+    original_get_move = tictactoe.GameInterface.get_move
+    tictactoe.GameInterface.get_move = lambda _: 'help'
+    test_file = open("test.txt", "w")
+    board = tictactoe.TicTacToe()
+    game = tictactoe.GameInterface(board, test_file)
+    game.player_move()
+    test_file.close()
+    output = open("test.txt", "r").read()
+
+    os.remove('test.txt')
+    tictactoe.GameInterface.get_move = original_get_move
+
+    assert (output == """
+    You can enter:
+    - 1 character from [abcdefghi] to place a mark in the corresponding square (eg "a")
+    - 2 characters from [abcdefghi] to place a split mark in corresponding squares (eg "bd")
+    - "map": show board map
+    - "exit" to quit
+    
+Still your move.
+"""
+    )
+    assert (game.player == 'X')
+
+def test_map():
+    original_get_move = tictactoe.GameInterface.get_move
+    test_file = open("test.txt", "w")
+    tictactoe.GameInterface.get_move = lambda _: 'map'
+    board = tictactoe.TicTacToe()
+    game = tictactoe.GameInterface(board, test_file)
+    game.player_move()
+    test_file.close()
+    output = open("test.txt", "r").read()
+    os.remove('test.txt')
+    tictactoe.GameInterface.get_move = original_get_move
+
+    assert (output == """
+        a | b | c
+        -----------
+        d | e | f
+        -----------
+        g | h | i
+        
+Still your move.
+"""
+    )
+    assert (game.player == 'X')
+
+def test_exit():
+    original_get_move = tictactoe.GameInterface.get_move
```

Comment:
Yikes.  This is super hacky.  Don't do this.  

Either mock get_move or modify it to provide a an input function as an argument, such as this (not quite sure if this exactly works as written but something like it):

```
    def get_move(self, input_fn: Callable[str,str]=input) -> str: 
        return input_fn(f"Player {self.player} to move (\"help\" for help): ")
```

Reply:
That's what I figured. I had tried using `@patch` decorators, but `MagicMock` ended up being much simpler.